web-search-engines/langchain4j-web-search-engine-searchapi/src/test/java/dev/langchain4j/web/search/searchapi/SearchApiWebSearchToolIT.java

```diff
@@ -0,0 +1,169 @@
+package dev.langchain4j.web.search.searchapi;
+
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.agent.tool.ToolSpecifications;
+import dev.langchain4j.data.message.*;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.model.openai.OpenAiChatModelName;
+import dev.langchain4j.service.AiServices;
+import dev.langchain4j.web.search.WebSearchEngine;
+import dev.langchain4j.web.search.WebSearchTool;
+import dev.langchain4j.web.search.WebSearchToolIT;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import static dev.langchain4j.web.search.searchapi.SearchApiWebSearchEngine.DEFAULT_ENV_VAR;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnabledIfEnvironmentVariable(named = "SEARCHAPI_API_KEY", matches = ".+")
+class SearchApiWebSearchToolIT extends WebSearchToolIT {
+		
+	static final boolean logRequests = true;
+
+    WebSearchEngine webSearchEngine = SearchApiWebSearchEngine
+    		.withApiKey(System.getenv(DEFAULT_ENV_VAR))
+    		.logRequests(logRequests)
+    		.build();
+
+    ChatLanguageModel chatModel = OpenAiChatModel.builder()
+            .apiKey(System.getenv("OPENAI_API_KEY"))
+            .modelName(OpenAiChatModelName.GPT_3_5_TURBO)
+            .logRequests(logRequests)
+            .build();
+
+    interface Assistant {
+        @dev.langchain4j.service.SystemMessage({
+                "You are a web search support agent.",
+                "If there is any event that has not happened yet",
+                "You MUST create a web search request with with user query and",
```

Comment:
your competitor had the same issue 😄 
```suggestion
                "You MUST create a web search request with user query and",
```

Reply:
Fixed.